test/PoolManager.t.sol

```diff
@@ -983,6 +985,15 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
         manager.burn(address(this), key.currency0.toId(), 1);
     }
 
+    function test_collectProtocolFees_locked_revertsWithProtocolFeeCurrencySynced() public {
+        assertNotEq(Currency.unwrap(key.currency1), address(0));
+        manager.sync(key.currency1);
+        assertEq(Currency.unwrap(key.currency1), Currency.unwrap(manager.getSyncedCurrency()));
```

Comment:
I added these assertions to debug the test failure. It seems this is the behavior that is different between `forge test` and `forge snapshot --isolate`:
```
[FAIL: assertion failed: 0x3Cff5E7eBecb676c3Cb602D0ef2d46710b88854E != 0x0000000000000000000000000000000000000000] test_collectProtocolFees_locked_revertsWithProtocolFeeCurrencySynced() (gas: 41905)
```
I do not know why `manager.sync` does not seem to work in `forge snapshot --isolate` but does work in `forge test`.

Reply:
ok I think I have a clue:

```
      --isolate
          Whether to enable isolation of calls. In isolation mode all top-level
          calls are executed as a separate transaction in a separate EVM
          context, enabling more precise gas accounting and transaction state
          changes
```

So this means I can only do one call per test.

Reply:
I wonder then, how does `test_collectProtocolFees_ERC20_accumulateFees_gas` work?

Reply:
> I wonder then, how does `test_collectProtocolFees_ERC20_accumulateFees_gas` work?

I think it works because this problem only impacts `TSTORE` and `TLOAD`.

Reply:
e2af146

Reply:
Therefore it's slightly harder to test the revert for the locked case, because I would need to write a batcher to consolidate both actions into the same transaction.